### PR TITLE
lui-track-bar: option to use a fringe indicator instead of a bar

### PR DESCRIPTION
**Pull request by jaor**
And a new command to jump to last unread line.
#

### CODE REVIEW

**Comment by wasamasa** [comment]
Thanks! One thing bothering me about this is that with this change the current name of the module becomes misleading because it's no longer necessarily a bar that's being displayed. Another is that this change assumes that there will be only two ways of signalling new messages since the last time you checked, it would be better for future extension to instead have a customizable holding a symbol of which way is currently enabled and a `cond` dispatching on that symbol (as opposed to the proposed boolean variable and an `if` dispatching on it).

**Comment by jaor** [comment]
your comments sound fair enough to me (even though i cannot think at the
moment of other ways of signalling).  how about renaming lui-track-bar
to simply lui-track?

**Comment by wasamasa** [comment]
That could work. In any case, some measures are needed to preserve backwards compatibility, like creating aliases of the old variables/commands/functions

**Comment by jaor** [comment]
yes.  please, let me know if/when you'd like me to do that. thanks.

**Comment by jaor** [comment]
i just went ahead and created lui-track. thanks.

File: lui-track-bar.el [hunk 1]
```diff
@@ -33,97 +33,25 @@
...
-      (lui-track-bar-move)))
-  (defadvice tracking-next-buffer (before lui-track-bar activate)
-    (when (eq lui-track-bar-behavior 'before-tracking-next-buffer)
-      (lui-track-bar-move)))
-  (add-hook 'lui-pre-input-hook 'lui-track-bar--move-pre-input))
+  (setq lui-track-indicator 'bar)
+  (enable-lui-track))
 
-;;;###autoload
-(defun lui-track-bar-jump-to-first-unread-line ()
```
Comment (wasamasa) [inline lui-track-bar.el]: I think that's one autoload not accounted for with an alias. But then, it doesn't seem like anyone uses it anyway.

File: lui-track.el [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      (lui-track-move)))
+  (add-hook 'lui-pre-input-hook 'lui-track--move-pre-input))
+
+;;;###autoload
+(defun lui-track-jump-to-indicator ()
+  "Move the point to the first unread line in this buffer.
+
+If point is already there, jump back to the end of the buffer."
+  (interactive)
+  (let ((ipos (case lui-track-indicator
```
Comment (wasamasa) [inline lui-track.el]: Use `cl-case` or a regular `cond` if you don't want to load up `cl-lib`.

File: lui-track.el [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+(defun lui-track--move-pre-input ()
+  (when (eq lui-track-behavior 'after-send)
+    (lui-track-move)))
+
+(defun lui-track-move ()
+  "Move the track indicator down."
+  (interactive)
+  (when (derived-mode-p 'lui-mode)
+    (case lui-track-indicator
```
Comment (wasamasa) [inline lui-track.el]: Same here

**Comment by jaor** [comment]
not totally sure i understand.  this function is new in this PR, so it's
normal nobody used it :)

**Comment by jaor** [comment]
just curious: what's wrong with `case`? it's just an alias for `cl-case`...

**Comment by wasamasa** [comment]
This is what the rest of the codebase happens to do. That aside it's an official recommendation, to quote the Emacs Lisp reference:

> • If you need Common Lisp extensions, use the ‘cl-lib’ library rather
> than the old ‘cl’ library.  The latter does not use a clean
> namespace (i.e., its definitions do not start with a ‘cl-’ prefix).
> If your package loads ‘cl’ at run time, that could cause name
> clashes for users who don’t use that package.
>
> There is no problem with using the ‘cl’ package at _compile_ time,
> with ‘(eval-when-compile (require 'cl))’.  That’s sufficient for
> using the macros in the ‘cl’ package, because the compiler expands
> them before generating the byte-code.  It is still better to use
> the more modern ‘cl-lib’ in this case, though.

I've finally got around to test your changes and everything works fine, both in the case of my config unaware of the change and my config adjusted to make use of the new name and customizable. For this reason I've merged it as is and added some adjustments to mark the old functions and variables obsolete. Thanks for your patience and contribution!

**Comment by jaor** [comment]
thanks a lot, very glad to contribute!

**Pull request merged**